### PR TITLE
added isEU

### PR DESCRIPTION
**Pull request by rm-Umar**

### CODE REVIEW

File: src/IPinfo/Models/IPResponse.cs [hunk 1]
```diff
@@ -17,6 +17,8 @@ public class IPResponse
       public string Country { get; private set; }
       
       public string CountryName { get; internal set; }
+
+      public bool isEU { get; internal set; }
```
Comment (UmanShahzad) [inline src/IPinfo/Models/IPResponse.cs]: Think it has to be `IsEU` here? (capital `I`).

File: src/IPinfo/Utilities/CountryHelper.cs [hunk 2]
```diff
@@ -12,6 +12,7 @@ namespace IPinfo.Utilities
     internal static class CountryHelper
     {
         private const string ConutriesJsonFilePathName = "IPinfo.Utilities.Countries.json";
+        private const string EUconutriesJsonFilePathName = "IPinfo.Utilities.EUcountries.json";
```
Comment (UmanShahzad) [inline src/IPinfo/Utilities/CountryHelper.cs]: Can we rename the file `EuCountries`? Looks weird that `countries` is lowercase like this.

File: src/IPinfo/Utilities/CountryHelper.cs [hunk 3]
```diff
@@ -30,10 +31,32 @@ internal static class CountryHelper
             return countries;
         });
 
+        /// <summary>
+        /// Lazy initialization for the EUcountry List from EUcountry json file.
+        /// </summary>
+        private static readonly Lazy<List<string>> e_countries =
```
Comment (UmanShahzad) [inline src/IPinfo/Utilities/CountryHelper.cs]:
```suggestion
        private static readonly Lazy<List<string>> eu_countries =
```

File: src/IPinfo/Utilities/CountryHelper.cs [hunk 4]
```diff
@@ -30,10 +31,32 @@ internal static class CountryHelper
...
+        new Lazy<List<string>>(() =>
+        {
+            List<string> eucountries;
```
Comment (UmanShahzad) [inline src/IPinfo/Utilities/CountryHelper.cs]:
```suggestion
            List<string> _eu_countries;
```

File: src/IPinfo/Utilities/CountryHelper.cs [hunk 5]
```diff
@@ -30,10 +31,32 @@ internal static class CountryHelper
...
+            var assembly = Assembly.GetExecutingAssembly();
+            using (Stream stream = assembly.GetManifestResourceStream(EUconutriesJsonFilePathName))
```
Comment (UmanShahzad) [inline src/IPinfo/Utilities/CountryHelper.cs]: Woah, lots of typos and weird naming schemes going on here! `countries`, not `conutries`, and `euCountriesJsonFilePathName`.

File: src/IPinfo/Utilities/CountryHelper.cs [hunk 6]
```diff
@@ -30,10 +31,32 @@ internal static class CountryHelper
...
+            using (StreamReader reader = new StreamReader(stream))
+            {
+                string eucountriesJson = reader.ReadToEnd();
```
Comment (UmanShahzad) [inline src/IPinfo/Utilities/CountryHelper.cs]: euCountriesJson

File: src/IPinfo/Utilities/CountryHelper.cs [hunk 7]
```diff
@@ -30,10 +31,32 @@ internal static class CountryHelper
...
+
         /// <summary>
         /// Dictionary for country_code to country_name mapping.
         /// </summary>
         private static Dictionary<string, string>  Countries { get { return s_countries.Value; } }
+        
+        /// <summary>
+        /// List of EU countries.
+        /// </summary>
+        private static List<string>  EUcountries { get { return e_countries.Value; } }
```
Comment (UmanShahzad) [inline src/IPinfo/Utilities/CountryHelper.cs]: `EuCountries` I guess?

File: src/IPinfo/Utilities/CountryHelper.cs [hunk 8]
```diff
@@ -56,5 +79,19 @@ internal static string GetCountry(string countryCode)
                 return countryCode;
             }
         }
+
+        /// <summary>
+        /// whether a country is a member of the EU or not.
+        /// </summary>
+        /// <param name="countryCode">Country code consisting of two characters.</param>
+        /// <returns>True if the country is a member of the European Union (EU) else false.</returns>
+        internal static bool isEU(string countryCode)
```
Comment (UmanShahzad) [inline src/IPinfo/Utilities/CountryHelper.cs]: IsEU in this language?

**Review by UmanShahzad — CHANGES_REQUESTED** [review]

**Comment by UmanShahzad** [comment]
@SobanMahmood can you double-check and confirm if it's good?

File: src/IPinfo/IPinfo.csproj [hunk 9]
```diff
@@ -38,6 +38,12 @@
     <PackageReference Include="System.Text.Json" Version="6.0.2" />
   </ItemGroup>
 
+  <ItemGroup>
+    <EmbeddedResource Include="Utilities\EuCountries.json" >
```
Comment (SobanMahmood) [inline src/IPinfo/IPinfo.csproj]: `EuCountries.json` can be replaced with `EUCountries.json` to be compatible with existing naming convention.

File: src/IPinfo/Utilities/CountryHelper.cs [hunk 10]
```diff
@@ -11,7 +11,8 @@ namespace IPinfo.Utilities
     /// </summary>
     internal static class CountryHelper
     {
-        private const string ConutriesJsonFilePathName = "IPinfo.Utilities.Countries.json";
+        private const string CountriesJsonFilePathName = "IPinfo.Utilities.Countries.json";
+        private const string EuCountriesJsonFilePathName = "IPinfo.Utilities.EuCountries.json";
```
Comment (SobanMahmood) [inline src/IPinfo/Utilities/CountryHelper.cs]: Same for naming convention, `EuCountriesJsonFilePathName` to be replaced with `EUCountriesJsonFilePathName`, and `IPinfo.Utilities.EuCountries.json` to be replaced with `IPinfo.Utilities.EUCountries.json`

File: src/IPinfo/Utilities/CountryHelper.cs [hunk 11]
```diff
@@ -30,10 +31,32 @@ internal static class CountryHelper
...
 
+        /// <summary>
+        /// Lazy initialization for the EUcountry List from EUcountry json file.
+        /// </summary>
+        private static readonly Lazy<List<string>> eu_countries =
+        new Lazy<List<string>>(() =>
+        {
+            List<string> _eu_countries;
+            var assembly = Assembly.GetExecutingAssembly();
+            using (Stream stream = assembly.GetManifestResourceStream(EuCountriesJsonFilePathName))
```
Comment (SobanMahmood) [inline src/IPinfo/Utilities/CountryHelper.cs]: `EuCountriesJsonFilePathName` to be replaced with `EUCountriesJsonFilePathName`.

File: src/IPinfo/Utilities/CountryHelper.cs [hunk 12]
```diff
@@ -30,10 +31,32 @@ internal static class CountryHelper
             return countries;
         });
 
+        /// <summary>
+        /// Lazy initialization for the EUcountry List from EUcountry json file.
+        /// </summary>
+        private static readonly Lazy<List<string>> eu_countries =
```
Comment (SobanMahmood) [inline src/IPinfo/Utilities/CountryHelper.cs]: Just to be on consistent convention for [private static](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions#camel-case), `eu_countries` to be replaced with `s_euCountries`.

File: src/IPinfo/Utilities/CountryHelper.cs [hunk 13]
```diff
@@ -30,10 +31,32 @@ internal static class CountryHelper
...
+        new Lazy<List<string>>(() =>
+        {
+            List<string> _eu_countries;
```
Comment (SobanMahmood) [inline src/IPinfo/Utilities/CountryHelper.cs]: `_eu_countries` to be replaced with `euCountries`.

File: src/IPinfo/Utilities/CountryHelper.cs [hunk 14]
```diff
@@ -30,10 +31,32 @@ internal static class CountryHelper
...
+            var assembly = Assembly.GetExecutingAssembly();
+            using (Stream stream = assembly.GetManifestResourceStream(EuCountriesJsonFilePathName))
+            using (StreamReader reader = new StreamReader(stream))
+            {
+                string euCountriesJson = reader.ReadToEnd();
+                _eu_countries = JsonSerializer.Deserialize<List<string>>(euCountriesJson);
```
Comment (SobanMahmood) [inline src/IPinfo/Utilities/CountryHelper.cs]: `_eu_countries` to be replaced with `euCountries`.

File: src/IPinfo/Utilities/CountryHelper.cs [hunk 15]
```diff
@@ -30,10 +31,32 @@ internal static class CountryHelper
...
+            }
+            return _eu_countries;
```
Comment (SobanMahmood) [inline src/IPinfo/Utilities/CountryHelper.cs]: `_eu_countries` to be replaced with `euCountries`.

File: src/IPinfo/Utilities/CountryHelper.cs [hunk 16]
```diff
@@ -30,10 +31,32 @@ internal static class CountryHelper
...
+
         /// <summary>
         /// Dictionary for country_code to country_name mapping.
         /// </summary>
         private static Dictionary<string, string>  Countries { get { return s_countries.Value; } }
+        
+        /// <summary>
+        /// List of EU countries.
+        /// </summary>
+        private static List<string>  EuCountries { get { return eu_countries.Value; } }
```
Comment (SobanMahmood) [inline src/IPinfo/Utilities/CountryHelper.cs]: `EuCountries` to be replaced with `EUCountries`.

File: src/IPinfo/Utilities/CountryHelper.cs [hunk 17]
```diff
@@ -56,5 +79,19 @@ internal static string GetCountry(string countryCode)
...
+        /// </summary>
+        /// <param name="countryCode">Country code consisting of two characters.</param>
+        /// <returns>True if the country is a member of the European Union (EU) else false.</returns>
+        internal static bool IsEU(string countryCode)
+        {
+            if(countryCode == null)
+            {
+                return false;
+            }
+            return EuCountries.Contains(countryCode);
```
Comment (SobanMahmood) [inline src/IPinfo/Utilities/CountryHelper.cs]: `EuCountries` to be replaced with `EUCountries`

**Review by SobanMahmood — CHANGES_REQUESTED** [review]
Just minor naming convention.

**Review by UmanShahzad — APPROVED** [review]
@SobanMahmood can you do one final check? @rm-Umar can you run test cases once?

After we merge, @SobanMahmood will need your help shipping this to the package managers.

**Review by SobanMahmood — APPROVED** [review]

**Pull request merged**